### ISSUE TITLE
InternalServerErrorException: Tracing Failed when tracing Sascha Willems particlefire demo on Linux

### ISSUE THREAD

**Issue by charlie-ht** [google/agi#370]
**Environment information:**
 - AGI version: `ffcead0e9616e3e9488596930d6192f1172cc9a9`
 - Host OS: Linux

**Bug description**

**Reproduction steps**
Run the particlefire demo from https://github.com/SaschaWillems/Vulkan repository in the Capture new trace flow with this configuration,
 
```
buffers {
  size_kb: 131072
  fill_policy: DISCARD
}
buffers {
  size_kb: 4096
  fill_policy: DISCARD
}
data_sources {
  config {
    name: "linux.ftrace"
    ftrace_config {
      ftrace_events: "sched/sched_process_free"
      ftrace_events: "task/task_newtask"
      ftrace_events: "task/task_rename"
      ftrace_events: "power/gpu_frequency"
      buffer_size_kb: 8192
      drain_period_ms: 250
      compact_sched {
        enabled: true
      }
    }
  }
}
data_sources {
  config {
    name: "linux.process_stats"
    target_buffer: 1
    process_stats_config {
      scan_all_processes_on_start: true
    }
  }
}
data_sources {
  config {
    name: "linux.process_stats"
    process_stats_config {
      proc_stats_poll_ms: 2000
      proc_stats_cache_ttl_ms: 20000
    }
  }
}
data_sources {
  config {
    name: "VulkanCPUTiming"
    legacy_config: "VkDevice:VkPhysicalDevice:VkInstance:VkQueue"
  }
}
data_sources {
  config {
    name: "VulkanMemoryTracker"
    legacy_config: "Device:Driver"
  }
}
write_into_file: true
file_write_period_ms: 2000
max_file_size_bytes: 2147483648
flush_period_ms: 5000
```

**Stacktrace**
```
com.google.gapid.server.Client$InternalServerErrorException: Tracing Failed
   Cause: exit status 1
	at com.google.gapid.server.Client.throwIfError(Client.java:310)
	at com.google.gapid.server.Client.lambda$streamTrace$55(Client.java:284)
	at com.google.gapid.server.GapidClientGrpc$StreamHandler.onNext(GapidClientGrpc.java:225)
	at io.grpc.stub.ClientCalls$StreamObserverToCallListenerAdapter.onMessage(ClientCalls.java:436)
	at io.grpc.internal.ClientCallImpl$ClientStreamListenerImpl$1MessagesAvailable.runInternal(ClientCallImpl.java:658)
	at io.grpc.internal.ClientCallImpl$ClientStreamListenerImpl$1MessagesAvailable.runInContext(ClientCallImpl.java:643)
	at io.grpc.internal.ContextRunnable.run(ContextRunnable.java:37)
	at io.grpc.internal.SerializingExecutor.run(SerializingExecutor.java:123)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628)
	at java.base/java.lang.Thread.run(Thread.java:834)
Caused by: com.google.gapid.server.Client$Stack: For request: RPC->streamTrace()
	at com.google.gapid.server.Client.streamTrace(Client.java:281)
	at com.google.gapid.server.Tracer.trace(Tracer.java:41)
	at com.google.gapid.views.TracerDialog.showTracingDialog(TracerDialog.java:176)
	at com.google.gapid.LoadingScreen.lambda$createOptions$0(LoadingScreen.java:119)
	at org.eclipse.swt.widgets.EventTable.sendEvent(EventTable.java:89)
	at org.eclipse.swt.widgets.Display.sendEvent(Display.java:5687)
	at org.eclipse.swt.widgets.Widget.sendEvent(Widget.java:1423)
	at org.eclipse.swt.widgets.Display.runDeferredEvents(Display.java:4955)
	at org.eclipse.swt.widgets.Display.readAndDispatch(Display.java:4448)
	at org.eclipse.jface.window.Window.runEventLoop(Window.java:823)
	at org.eclipse.jface.window.Window.open(Window.java:799)
	at com.google.gapid.Main$UI.show(Main.java:128)
	at com.google.gapid.Main.main(Main.java:77)
```

**Screenshots**
![Step 1, Capture System Profile](https://user-images.githubusercontent.com/14251608/86785535-b08f6a80-c05a-11ea-9bf5-dc2b7bde17d1.png)
![Step 2. particlefire starts, and AGI waits for me to click Start](https://user-images.githubusercontent.com/14251608/86785641-da489180-c05a-11ea-98ab-f294cb87afce.png)
![Step 3. AGI fails with stacktrace above](https://user-images.githubusercontent.com/14251608/86785729-f64c3300-c05a-11ea-8afe-af546ddb24ac.png)

**Additional debugging information**
- Please attach the generated _gapis.log_ and _gapic.log_ files you will find in the temp folder (e.g. /tmp/ on linux).
[gapis.log](https://github.com/google/agi/files/4885007/gapis.log) 
[gapic.log](https://github.com/google/agi/files/4885022/gapic.log)

The real stuff starts towards the bottom of the file, after me finding the adb path. It seems to connect to perfetto service, and then immediately hangup with no obvious reason (to me)

**Comment by hevrard** [comment]
@charlie-ht thanks for reporting!

I'm afraid profiling is only available on [supported Android devices](https://gpuinspector.dev/docs/requirements#supported-android-devices), with currently the additional requirement of special preview GPU drivers available through the developer preview program (see https://gpuinspector.dev/). So AGI won't be able to profile on your laptop Intel GPU.

Now, I'm surprised that AGI even reports your Intel GPU as "Validation Passed", and that it offers the possibility to do a system profile. Did you get AGI to list your Intel GPU as a valid device for system profiling out of the box?

(That would be a bug, AGI is not supposed to report this GPU as a valid one for profiling).

**Comment by silence-do-good** [comment]
@charlie-ht Did you by any chance run a perfetto trace from the perfetto standalone sdk and then try this? AGI checks for a temp file (/tmp/perfetto-consumer) to see if a device supports system profiling. This file gets created when you run traced (either directly or by using the "tools/tmux" command from [https://perfetto.dev/docs/quickstart/linux-tracing](url)). If this is indeed the case, then traced is active only for the time you specified in the config while running the perfetto trace.
The gapis.log indicates that it cannot connect to traced, meaning traced is not alive.

In short, like @hevrard said, the main targets for system profiling are the supported Android devices. If you still want to test the desktop tracing, I'd suggest you start the traced and then try to take the trace with AGI.

**Comment by charlie-ht** [comment]
Thank you @hevrard & @silence-do-good, I did read about the supported Android devices, but thought I'd give it a try anyway on my laptop. I'd like an integrated profiling solution on Linux too, then I could modify some open-source driver to instrument it with trace calls to Perfetto. I think this is what happens in the Android GPU drivers inside `libgpudataproducer.so` from my skiming of the codebase. Do you know of any work in this direction?

@silence-do-good You are absolutely correct! I verified that when I remove these stale files from previous experiments, AGI does not let me try and trace. Mystery solved. Let me know if you'd like any further information, but it sounds like you know what is the problem :-)

**Comment by pmuetschard** [comment]
Looks like there is nothing left to do here. To trace on desktop, it is (currently) the user's responsibility to run `traced` and `traced-probes` and ensure they are running properly before attempting to trace. These binaries are shipped as part of AGI, in the application's `perfetto` folder.
